torch/csrc/jit/frontend/tracer.h

```diff
@@ -355,6 +359,7 @@ TORCH_API void addOutput(
     Node* node,
     const c10::intrusive_ptr<c10::ivalue::Object>& output);
 
+TORCH_API void addOutput(Node* node, const at::TensorList outputs);
```

Comment:
why is this not captured by:
`TORCH_API void addOutput(Node* node, const std::vector<at::Tensor>& list);` ?

Reply:
Because TensorList is not a std::vector but an [ArrayRef](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/templates/TensorBody.h#L50)

in tracer.cpp im passing the given TensorList to this overload
```
void addOutput(Node* node, const at::TensorList outputs) {
  return addOutput(node, outputs.vec());
}
```

Reply:
how can you return an ArrayRef?

Reply:
I guess we know that the ArrayRef is valid through the entire path?

Reply:
I was actually making a different point.

The C++ signature is:
`std::vector<Tensor> foreach_add_scalar_kernel_fallback_(TensorList tensors, Scalar scalar) {`

which suggests a vector is an output, but here you need a TensorList as an output?

Reply:
As @ezyang points out, we don't need method chaining (we don't support it on TensorLists) and we do have mutable signatures that don't have returns:
https://github.com/pytorch/pytorch/blob/e8f4b04d9ae1880fd8d09dde40639e86a0d2c585/aten/src/ATen/native/native_functions.yaml#L1006

You might have to play around with some combinations to get the codegen correct, though.

Reply:
Done!